packages/doc/docs/api-building/sql-template.mdx

```diff
@@ -0,0 +1,413 @@
+
+```sql
+# Display the variable value
+{% set myArray = [1,2,3,4,4] %}
+SELECT {{ myArray }}
+
+------
+# The response by sending the request which used curl command
+< HTTP/1.1 200 OK
+< Vary: Origin
+< ...
+[{"$1":"1,2,3,4,4"}]%
+```
+
+The API sends the response and value because it’s a SQL File, so when you use `SELECT` clause, it could get the result.
+
+But you will see the field is `$1` in the response, if you would like to define the key, you can use `AS` (alias) for giving the column name, and it will become the field name.
+
+```sql
+# Display the variable value
+{% set myArray = [1,2,3,4,4] %}
+SELECT {{ myArray }} as arr
+
+------
+# The response by sending the request which used curl command
+< HTTP/1.1 200 OK
+< Vary: Origin
+< ...
+[{"arr":"1,2,3,4,4"}]%
+```
+
+Otherwise, you will see the above example always print the value to `$[number]` e.g: `$1` and then print the value in the terminal and display the `$1` as key and then shows the value.
+
+The reason is that VulcanSQL **_uses the placeholder_** and query by **_parameterized statement for preventing SQL injection._**
+
+You could also print the dynamic parameter in the terminal too ( If you would like to get a response, still add the `SELECT` in the head:
+
+```sql
+
+# Display the dynamic parameter value
+# context.params.age = 18
+{% set myAge = context.params.age %}
+{{ myAge }} # -> $1 -> 18
+---
+# Display the dynamic parameter value which added primitive integer
+# context.params.age = 18
+{% set myAge = (context.params.age + 1) %}
+{{ myAge }} # -> $1 -> 19
+
+```
+
+## Filters
+
+Filters are one of the [nunjucks](https://mozilla.github.io/nunjucks/templating.html#tags) features which can apply the function to variables by pipe operator (`|`) and can take arguments, shown below:
+
+```sql
+# context.params.name is 'canner'
+{{ context.params.name | upper }} # -> $1 -> 'CANNER'
+```
+
+Of course, you could use the `SELECT` to see the response:
+
+```sql
+# context.params.name is 'canner'
+SELECT {{ context.params.name | upper }} # -> $1 -> 'CANNER'
+
+----
+# The response by sending the request which used curl command
+< HTTP/1.1 200 OK
+< Vary: Origin
+< ...
+[{"$1":"CANNER"}]%
+```
+
+You could also use multiple filters:
+
+```sql
+{% set items = ['foo', 'bar', 'bear'] %}
+SELECT {{ items | join(",") | upper }} # -> $1 -> 'FOO,BAR,BEAR'
+```
+
+You could also use the [nunjucks](https://mozilla.github.io/nunjucks/templating.html#tags) built-in Filters functions, please see the [nunjucks filters](https://mozilla.github.io/nunjucks/templating.html#builtin-filters).
+
+### Raw Filter
+
+The `raw` filter is the VulcanSQL custom filer function, which could make your defined variable or dynamic parameter output value directly, it will transform `$[number]` to the value first and if you still have a pipe function after the `raw` , then it will use the output value to do the filter.
```

Comment:
Same as here, could you assist me to change the `$[number]` to `$<number>`, thanks ?